### PR TITLE
fix(core): core time-picker accessibility

### PR DESCRIPTION
**Pull request by dimamarksman**
## Related Issue(s)

closes https://github.com/sap/fundamental-ngx/issues/6182

## Description

This PR fixes some accessibility issues with in core time-picker

##### During Implementation

1. Visual Testing:

-   [x] visual misalignments/updates
-   [x] check Light/Dark/HCB/HCW themes
-   [x] RTL/LTR - proper rendering and labeling
-   [x] responsiveness(resize)
-   [x] Content Density (Cozy/Compact/(Condensed))
-   [x] States - hover/disabled/focused/active/on click/selected/selected hover/press state
-   [x] Interaction/Animation - open/close, expand/collapse, add/remove, check/uncheck
-   [x] Mouse vs. Keyboard support
-   [x] Text Truncation

2. API and functional correctness

-   [x] check for console logs (warnings, errors)
-   [x] API boundary values
-   [x] different combinations of components - free style
-   [x] change the API values during testing

3. Documentation and Example validations

-   [na] missing API documentation or it is not understandable
-   [na] poor examples
-   [x] Stackblitz works for all examples

4. Accessibility testing
5. Browser Testing - Edge, Safari, Chrome, Firefox

##### PR Quality

-   [x] the commit message(s) follows the guideline:
        https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
-   [x] tests for the changes that have been done
-   [x] all items on the PR Review Checklist are addressed :
        https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
-   [x] Run npm run build-pack-library and test in external application
-   [na] update `README.md`
-   [na] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)

### CODE REVIEW

**Comment by dimamarksman** [comment]
blocked until https://github.com/SAP/fundamental-ngx/pull/7280 is merged

**Comment by dimamarksman** [comment]
Issue 3 with the backslash is not fixed. Please see [my comment here](https://github.com/sap/fundamental-ngx/issues/6182#issuecomment-984515503) 
Issue 4 can not be reproduced.

File: libs/core/src/lib/shared/services/value-state-message.service.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    readonly success = 'Value State Success';
+
+    /** Value state "information" message */
+    readonly information = 'Value State Information';
+
+    /** Value state "warning" message */
+    readonly warning = 'Value State Warning';
+
+    /** Value state "error" message */
+    readonly error = 'Value State Error';
```
Comment (InnaAtanasova) [inline libs/core/src/lib/shared/services/value-state-message.service.ts:27]: How are we handling i18n?

**Review by InnaAtanasova — COMMENTED** [review]
Just one question I added for i18n, the rest looks good

File: libs/core/src/lib/shared/services/value-state-message.service.ts (hunk 1 above)
Comment (dimamarksman) [reply libs/core/src/lib/shared/services/value-state-message.service.ts:27]: This can be overridden on app level with translated strings

Comment (mikerodonnell89) [reply libs/core/src/lib/shared/services/value-state-message.service.ts:27]: This goes against our current paradigm of allowing the devs to provide translations via inputs, this could be confusing

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]
see comment above re: i18n

File: libs/core/src/lib/shared/services/value-state-message.service.ts (hunk 1 above)
Comment (dimamarksman) [reply libs/core/src/lib/shared/services/value-state-message.service.ts:27]: @mikerodonnell89 there are still inputs for these fields which take precedence over these service values.
This service gives us default values for this kind of inputs.
So if app developer wants to redefine default values for components that rely on this it is needed only to override this service and not inputs for every occurrence in the template

Comment (dimamarksman) [reply libs/core/src/lib/shared/services/value-state-message.service.ts:27]: we have the same approach for core calendar labels

File: nx.json [hunk 2]
```diff
@@ -486,7 +486,7 @@
         },
         "core-time-picker": {
             "tags": ["fd"],
-            "implicitDependencies": ["core-utils", "core-time", "core-popover"]
+            "implicitDependencies": ["core-shared", "core-utils", "core-time", "core-popover"]
```
Comment (dmitry-stepanenko) [inline nx.json:489]: Why is this needed? In PR for Angular 13 migration all implicit dependencies for core packages are being removed now

Comment (dimamarksman) [reply nx.json:489]: the changes you are referencing are not merged yet

Comment (dimamarksman) [reply nx.json]: reverted it

**Review by dmitry-stepanenko — APPROVED** [review]

File: libs/core/src/lib/time-picker/time-picker.component.ts [hunk 3]
```diff
@@ -220,6 +226,30 @@ export class TimePickerComponent<D>
     @Input()
     parseFormat: unknown;
 
+    /**
+     * Value state "success" aria message.
+     */
+    @Input()
+    valueStateSuccessMessage: string = this._valueStateAriaMessagesService.success;
```
Comment (platon-rov) [inline libs/core/src/lib/time-picker/time-picker.component.ts:233]: As `_valueStateAriaMessagesService` is Optional so please use `?.` for optional chaining here and below.

Comment (dimamarksman) [reply libs/core/src/lib/time-picker/time-picker.component.ts:233]: ValueStateAriaMessageService is not optional
The service itself uses `provideIn: root`

**Review by Betrozov — APPROVED** [review]

**Review by InnaAtanasova — APPROVED** [review]

**Comment by dimamarksman** [comment]
@george861 @Sanjay-1312 could you please test it

**Comment by Sanjay-1312** [comment]
Issue 1: Fixed.

Issue 2: Fixed.

Issue 3: Fixed.

Issue 4: Fixed.

Retest Details:
https://deploy-preview-7304--fundamental-ngx.netlify.app/#/core/time-picker
JAWS 2021
